go/adbc/validation/validation.go

```diff
@@ -102,6 +102,33 @@ func (d *DatabaseTests) TestNewDatabase() {
 	d.Implements((*adbc.Database)(nil), db)
 }
 
+func (d *DatabaseTests) TestNewDatabaseGetSetOptions() {
```

Comment:
Ah, it seems this test should get moved into the Snowflake tests specifically? Other drivers don't allow arbitrary options

Reply:
Sure, I can do that. This might be testing a small enough edge case anyway that other drivers would be unlikely to benefit from the coverage.

Reply:
Updated